From 8f9f05866433050d308410d2649c1faa2471d199 Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Wed, 23 Aug 2023 14:45:49 +0200
Subject: [PATCH] build: don't fail when git-metadata is unavailable

This allows downstream distributions to builds from tarballs which lack
any git metadata.

See: https://github.com/ammen99/wf-recorder/pull/226
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 42ac49f..8e5a179 100644
--- a/meson.build
+++ b/meson.build
@@ -24,8 +24,8 @@ conf_data.set('fallback_audio_sample_fmt', get_option('fallback_audio_sample_fmt
 version = '"@0@"'.format(meson.project_version())
 git = find_program('git', native: true, required: false)
 if git.found()
-	git_commit = run_command([git, 'rev-parse', '--short', 'HEAD'], check: true)
-	git_branch = run_command([git, 'rev-parse', '--abbrev-ref', 'HEAD'], check: true)
+	git_commit = run_command([git, 'rev-parse', '--short', 'HEAD'], check: false)
+	git_branch = run_command([git, 'rev-parse', '--abbrev-ref', 'HEAD'], check: false)
 	if git_commit.returncode() == 0 and git_branch.returncode() == 0
 		version = '"@0@-@1@ (" __DATE__ ", branch \'@2@\')"'.format(
 			meson.project_version(),
-- 
2.42.0

